client/src/import_enhancement_provider.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  CompletionItemProvider,
+  TextDocument,
+  Position,
+  CompletionItem,
+  Disposable,
+  CompletionItemKind,
+  CompletionList,
+  DocumentSelector,
+  languages,
+  ExtensionContext,
+  Range,
+  Command,
+} from "vscode";
+
+import {
+  listVersionsOfMod,
+  modTreeOf,
+  parseImportStatement,
+} from "./import_utils";
+
+export class ImportEnhancementCompletionProvider
+  implements CompletionItemProvider, Disposable {
+  async provideCompletionItems(
+    document: TextDocument,
+    position: Position
+    // _token: CancellationToken,
+    // _context: CompletionContext
+  ): Promise<CompletionItem[] | CompletionList | undefined> {
+    const line_text = document.lineAt(position).text;
+
+    if (/import.+?from\W+['"].*?['"]/.test(line_text)) {
+      // We're at import statement line
+      const imp_info = parseImportStatement(line_text);
+      if (
+        imp_info?.domain !== "deno.land" ||
+        imp_info.module === undefined ||
+        imp_info.module.length === 0
+      ) {
+        return undefined;
+      }
+      // We'll handle the completion only if the domain is `deno.land` and mod name is not empty
+      const current_char = line_text[position.character - 1];
+      if (current_char === "@") {
```

Comment:
It would be great if we could do this based on your cursor position in the import statement, rather than checking that the previous char is `@`. It would be great if the autocomplete worked anywhere between the `@` for the version and the `/` of the path. Same goes for the module name. It should work anywhere between the `/x/` and `@`.

Reply:
>  Same goes for the module name. It should work anywhere between the /x/ and @.
module already works that way, because search result is not alphabetical order, the completion does not work well, but you can always hit CTRL+SPACE to retrigger search
the version one I'll change later

Reply:
@lucacasonato I don't know how to resolve [this isssue](https://github.com/microsoft/vscode/issues/106090), so I'm gonna leave it right now. And I think it's ready for merge

Reply:
SGTM :+1: